lib/cisco_node_utils/snmpuser.rb

```diff
@@ -74,32 +74,39 @@ def initialize_validator(name, groups, authproto, authpass, privproto,
       end
     end
 
-    ENGINE_ID_PATTERN = /([0-9]{1,3}(:[0-9]{1,3}){4,31})/
     def self.users
       users_hash = {}
       # config_get returns hash if 1 user, array if multiple, nil if none
       users = config_get('snmp_user', 'user')
       return users_hash if users.nil?
-      users = [users] if users.is_a?(Hash)
       users.each do |user|
-        name = user['user']
-        engineid = user['engineID']
-        if engineid.nil?
+        # n7k has enforcepriv, use-ipv*acl, avoid them
+        next if user[/(enforcePriv|use-ipv4acl|use-ipv6acl)/]
```

Comment:
I was told not to put any platform specific code in the ruby files. Since this is just one line I just put it there. Also these are not users, these are properties specific to n7k and they end up in the strings which we need to avoid. I can add a property in the yaml file to get the platform information and use it to filter this in the ruby file. Is this fine? If not, please send me an example on how to do it.